### DISCUSSION TITLE
How to export content, workflows, etc to recipe.json

### DISCUSSION THREAD

**Discussion by rwxzig** [OrchardCMS/OrchardCore#15119] [Q&A] [answered]
Is there any way of exporting everything that has been done through the GUI to a recipe in json so that it can be replicated next time easily?

**Comment by Piedone** [comment, marked as answer]
You need to [create a Deployment Plan](https://www.youtube.com/watch?v=2c5pbXuJJb0&list=PLuskKJW0FhJcSX7j0Bd-1X5hq3dgCtYwO&index=15) and add all the steps available.

I have this idea open to automated this, but it's not that simple: https://github.com/Lombiq/Helpful-Extensions/issues/105.

**Reply by rwxzig** [reply]
I get this error when I click to deployment plans:

> An unhandled exception occurred while processing the request.
InvalidOperationException: Unable to resolve service for type 'OrchardCore.Search.Elasticsearch.Core.Services.ElasticIndexSettingsService' while attempting to activate 'OrchardCore.Search.Elasticsearch.Core.Deployment.ElasticIndexResetDeploymentStepDriver'.

I am using version 1.8.2 could it be a bug?

Edit: I tried with version 1.7.2 and it works, but it seems like it is a bug in 1.8.2 version.

**Reply by Piedone** [reply]
That may be the case. @MikeAlhayek or @Skrypt any feedback on this?

**Reply by Skrypt** [reply]
It is possible that @MikeAlhayek broke it with the latest changes he made. Though, looks like an easy fix to do. Will try to repro.

**Reply by Skrypt** [reply]
Can't repro with latest dev branch. Please open an issue with repro steps.

**Reply by MikeAlhayek** [reply]
@cptzig it is likely that you enabled the Elasticsearch feature but you did not configure the elasticsearch services. Can you confirm that the settings for ES exists?

**Reply by MikeAlhayek** [reply]
@Skrypt I don't think it is an issue that introduced. ES always followed anti-discovery pattern. It is on my list to change but have not got to do it yet. these services are only available if the service is configured

**Reply by rwxzig** [reply]
@MikeAlhayek yes you are correct, I have enabled the Elastic search, and as for services configured, I have not configured any so we can safely assume that they are not available.

**Reply by MikeAlhayek** [reply]
@cptzig  @Skrypt  @Piedone  PR #15134 will prevent this issue in the future.
